extensions/ql-vscode/src/view/results/results.tsx

```diff
@@ -70,239 +71,236 @@ export const onNavigation = new EventHandlerList<NavigateMsg>();
 /**
  * A minimal state container for displaying results.
  */
-export class ResultsApp extends React.Component<
-  Record<string, never>,
-  ResultsViewState
-> {
-  constructor(props: any) {
-    super(props);
-    this.state = {
-      displayedResults: {
-        resultsInfo: null,
-        results: null,
-        errorMessage: "",
-      },
-      nextResultsInfo: null,
-      isExpectingResultsUpdate: true,
-    };
-  }
-
-  handleMessage(msg: IntoResultsViewMsg): void {
-    switch (msg.t) {
-      case "setState":
-        this.updateStateWithNewResultsInfo({
-          resultsPath: msg.resultsPath,
-          parsedResultSets: msg.parsedResultSets,
-          origResultsPaths: msg.origResultsPaths,
-          sortedResultsMap: new Map(Object.entries(msg.sortedResultsMap)),
-          database: msg.database,
-          interpretation: msg.interpretation,
-          shouldKeepOldResultsWhileRendering:
-            msg.shouldKeepOldResultsWhileRendering,
-          metadata: msg.metadata,
-          queryName: msg.queryName,
-          queryPath: msg.queryPath,
-        });
+export function ResultsApp() {
+  const [state, setState] = React.useState<ResultsViewState>({
```

Comment:
I opted to keep the state as one big object, instead of decomposing into three individual state objects that could be set separately. The reason for this is that in `updateStateWithNewResultsInfo` we set all three fields at once in a few different ways, so it's helpful to set them all together.

The only other place we update the state is in `handleMessage` and there we do only update `isExpectingResultsUpdate`, but thankfully we can do
```typescript
setState((prevState) => ({
  ...prevState,
  isExpectingResultsUpdate: true,
}));
```
so setting just one field and not the others is simple.

---

We could also have taken the option to have three separate state fields. If you call multiple `setState` handlers in a single render / event handler then React will only trigger a single re-render to handle all of the state changes. So performance-wise it would be exactly the same, but I think having them together is maybe slightly easier to code in this case.

I'm open to both options, so I'd be happy to separate them if people think that would be better. It's also worth nothing we aren't locked into this, and we could separate them in the future if that becomes a better option.